pandas/core/reshape/reshape.py

```diff
@@ -645,6 +645,10 @@ def _convert_level_number(level_num, columns):
         unique_groups = [key for key, _ in itertools.groupby(tuples)]
         new_names = this.columns.names[:-1]
         new_columns = MultiIndex.from_tuples(unique_groups, names=new_names)
+        new_columns = new_columns.set_levels([
```

Comment:
instead if u use MultiIndex.from_arrays then this step is incessant

Reply:
Done.